clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -10026,6 +10026,10 @@ def err_duplicate_case_differing_expr : Error<
 def warn_case_empty_range : Warning<"empty case range specified">;
 def warn_missing_case_for_condition :
   Warning<"no case matching constant switch condition '%0'">;
+def err_loop_attr_duplication : Error<
+  "duplicate loop attribute %0">;
+def err_attribute_power_of_two_in_range : Error<
+  "%0 attribute must be a constant power of two between %1 and %2 inclusive">;
```

Comment:
```suggestion
  "%0 attribute requires an integer argument which is a constant power of two between %1 and %2 inclusive%select{|; got %3}4">;
```
The idea here is that for non-integer constant expression you can say it requires the integer argument, and if you got an integer constant expression but it's out of range, then you'll add the "got <whatever>" part as well.

Reply:
I have updated diagnostic I have updated source code to pass the calculated alignment value https://github.com/llvm/llvm-project/pull/70762/commits/4c40ce27e81ae40f05aeabb513881032c04feb92